qir/qat/Generators/ProfileGenerator.cpp

```diff
@@ -50,35 +54,31 @@ Profile ProfileGenerator::newProfile(String const& name, OptimizationLevel const
             llvm::outs() << "Setting " << c.first << " up\n";
         }
 
-        c.second(this, ret);
-        module_pass_manager.addPass(FunctionToModule(std::move(function_pass_manager)));
+        c.second(*this, *ret);
+        module_pass_manager_->addPass(llvm::createModuleToFunctionPassAdaptor(std::move(function_pass_manager)));
+        // module_pass_manager_->addPass(FunctionToModule(std::move(function_pass_manager)));
```

Comment:
Can this commented out line be removed or matched with a TODO to give it context?